File: Samples/GridLoaders/DNN/VB/GridLoader/Navigation.ascx.vb
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Collections
Imports System.IO
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports DotNetNuke
Imports DotNetNuke.Common
Imports DotNetNuke.Entities.Modules
Imports DotNetNuke.Entities.Modules.Actions
Imports DotNetNuke.Entities.Users
Imports DotNetNuke.Security
Imports DotNetNuke.Services.Exceptions
Imports DotNetNuke.Services.Localization

Namespace EntitySpaces.Modules.GridLoader
    Partial Public Class Navigation
        Inherits PortalModuleBase
        ''' <summary>
        ''' Overide the page init event 
        ''' </summary>
        ''' <param name="e"></param>
        Protected Overloads Overrides Sub OnInit(ByVal e As EventArgs)
            MyBase.OnInit(e)

            Dim ModuleBase As PortalModuleBase = DirectCast((Parent.Parent), PortalModuleBase)
            If Not (ModuleBase Is Nothing) Then
                ModuleConfiguration = ModuleBase.ModuleConfiguration
            End If

            'LocalResourceFile = String.Concat(TemplateSourceDirectory, "/App_LocalResources/Index.ascx.resx")
        End Sub

        Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
            CreateIndex()
        End Sub

        Private Sub CreateIndex()
            Try
                Dim di As New DirectoryInfo(Server.MapPath(Constants.GeneratedControlsDirectory))
                Dim files As FileInfo() = di.GetFiles(String.Concat("*", Constants.ControlFileExtenstion), SearchOption.TopDirectoryOnly)

                For Each file As FileInfo In files
                    Dim link As New HyperLink()
                    link.NavigateUrl = Utilities.ConstructUrl(Constants.ControlKey, file.Name.Replace(String.Concat(Constants.ControlPostfix, Constants.ControlFileExtenstion), String.Empty))

                    link.Text = file.Name.Replace(String.Concat(Constants.ControlPostfix, Constants.ControlFileExtenstion), String.Empty)

                    NavigationMenuStripPlaceHolder.Controls.Add(link)
                    NavigationMenuStripPlaceHolder.Controls.Add(New LiteralControl("&nbsp;&nbsp;&nbsp;"))
                Next
            Catch exc As Exception

                Exceptions.ProcessModuleLoadException(Me, exc)
            End Try
        End Sub
    End Class
End Namespace

File: Samples/GridLoaders/DNN/C#/GridLoader/Navigation.ascx.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.IO;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using DotNetNuke;
using DotNetNuke.Common;
using DotNetNuke.Entities.Modules;
using DotNetNuke.Entities.Modules.Actions;
using DotNetNuke.Entities.Users;
using DotNetNuke.Security;
using DotNetNuke.Services.Exceptions;
using DotNetNuke.Services.Localization;

namespace EntitySpaces.Modules.GridLoader
{
    public partial class Navigation : PortalModuleBase
    {
        /// <summary>
        /// Overide the page init event 
        /// </summary>
        /// <param name="e"></param>
        protected override void OnInit(EventArgs e)
        {
            base.OnInit(e);

            PortalModuleBase ModuleBase = ((PortalModuleBase)(Parent.Parent));
            if (!(ModuleBase == null))
            {
                ModuleConfiguration = ModuleBase.ModuleConfiguration;
            }

            LocalResourceFile = string.Concat(TemplateSourceDirectory, "/App_LocalResources/Index.ascx.resx");
        }

        protected void Page_Load(object sender, EventArgs e)
        {
            CreateIndex();
        }

        private void CreateIndex()
        {
            try
            {
                DirectoryInfo di = new DirectoryInfo(Server.MapPath(Constants.GeneratedControlsDirectory));
                FileInfo[] files = di.GetFiles(string.Concat("*", Constants.ControlFileExtenstion), SearchOption.TopDirectoryOnly);

                foreach (FileInfo file in files)
                {
                    HyperLink link = new HyperLink();
                    link.NavigateUrl = Utilities.ConstructUrl(Constants.ControlKey, 
                      file.Name.Replace(string.Concat(Constants.ControlPostfix, Constants.ControlFileExtenstion), string.Empty));

                    link.Text = file.Name.Replace(
                        string.Concat(Constants.ControlPostfix, Constants.ControlFileExtenstion), string.Empty);

                    NavigationMenuStripPlaceHolder.Controls.Add(link);
                    NavigationMenuStripPlaceHolder.Controls.Add(new LiteralControl("&nbsp;&nbsp;&nbsp;"));
                }
            }

            catch (Exception exc)
            {
                Exceptions.ProcessModuleLoadException(this, exc);
            }
        }
    }
}